docs/api/modulus.distributed.shardtensor.rst

```diff
@@ -0,0 +1,117 @@
+        
+        # Redistribute to different sharding scheme
+        new_sharded = sharded.redistribute(
+            placements=(Shard(1),)  # Change to shard along second dimension
+        )
+
+How does this work?
+""""""""""""""""""
+
+ShardTensor extends PyTorch's DTensor to support uneven sharding where different ranks can have different 
+local tensor sizes. It tracks shard size information and handles redistribution between different 
+sharding schemes while maintaining gradient flow.
+
+Key differences from DTensor include:
+- Support for uneven sharding where ranks have different local sizes
+- Tracking and propagation of shard size information
+- Custom collective operations optimized for uneven sharding
+- Flexible redistribution between different sharding schemes
+
+Operations work by:
+1. Converting inputs to local tensors
+2. Performing operations locally 
+3. Constructing new ShardTensor with appropriate sharding
+4. Handling any needed communication between ranks
+
+.. autosummary::
+   :toctree: generated
+
+ShardTensor
+-----------
+
+.. autoclass:: modulus.distributed.shard_tensor.ShardTensor
+    :members:
+    :show-inheritance:
+
+Utility Functions
+----------------
+
+.. autofunction:: modulus.distributed.shard_tensor.scatter_tensor
+
+
+Why do we need this?
+""""""""""""""""""""
+
+During deep learning training, memory usage can grow significantly when working with large input data, even if the model itself is relatively small. This is because many operations create intermediate tensors that temporarily consume memory.
+
+For example, consider a 2D convolution operation on a high-resolution image. If we have a batch of 1024x1024 images, even a simple 3x3 convolution needs to save the entire input image in memory for computing the gradients in the backward pass.
+
+For high resolution images, this can easily lead to out of memory errors as model depth grows, even if the number of parameters is small - this is a significant contrast from LLM model training, where the memory usage is dominated by the number of parameters and the corresponding optimizer states.  In software solutions like DeepSpeed and ZeRO, this is handled by partitioning the model across GPUs, but this is not a solution for large-input applications.
+
+ShardTensor helps address this by:
+- Distributing the input data across multiple devices
+- Performing operations on smaller local portions
+- Coordinating the necessary communication between devices in the forward and backward passes
+
+ShardTensor is built as an extension of PyTorch's DTensor, and gains substantial functionality by leveraging the utilities already implemented in the PyTorch distributed package.  However, some operations on sharded input data are not trivial to implement correctly, nor relevant to the model sharding problem.  In Modulus, we have implemented parallelized versions of several key operations, including (so far):
+
+- Convolution (1D, 2D, 3D)
+- Neighborhood Attention (2D)
```

Comment:
Whether it's here or elsewhere, probably good to keep an official list of ops that are supported, and maybe link to them